lib/_tls_wrap.js

```diff
@@ -411,7 +411,8 @@ function TLSSocket(socket, opts) {
   net.Socket.call(this, {
     handle: this._wrapHandle(wrap),
     allowHalfOpen: socket ? socket.allowHalfOpen : tlsOptions.allowHalfOpen,
-    readable: false,
+    pauseOnCreate: tlsOptions.pauseOnConnect,
+    readable: tlsOptions.pauseOnConnect,
```

Comment:
Maybe a comment here? This is a little weird - if `readable` is false `pauseOnCreate` is not handled at all in `net.Socket`.

Reply:
```suggestion
    // readable only needed if pauseOnCreate is set
    readable: tlsOptions.pauseOnConnect,
```